station-items-v0/src/main/java/net/modificationstation/stationapi/api/dispenser/ItemDispenseContext.java

```diff
@@ -0,0 +1,133 @@
+package net.modificationstation.stationapi.api.dispenser;
+
+import net.minecraft.block.entity.DispenserBlockEntity;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.ItemEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+import net.modificationstation.stationapi.api.item.CustomDispenseBehavior;
+
+import java.util.Random;
+
+/**
+ * Utility class for changing dispenser behavior, providing tools to customize behavior
+ * @author matthewperiut
+ * @see DispenseEvent
+ * @see CustomDispenseBehavior
+ */
+public class ItemDispenseContext {
+    public final Random random = new Random();
+    public final World world;
+    public final DispenserBlockEntity dispenserBlockEntity;
+    public final ItemStack[] inventory;
+    public final int slot;
+    public final ItemStack itemStack;
+    private ItemStack shotItemStack;
+    public byte xDir = 0;
+    public byte zDir = 0;
+    public int x;
+    public int y;
+    public int z;
+    public double xVel;
+    public double yVel;
+    public double zVel;
```

Comment:
What's the point of making these six non-final if they're always calculated to the same value?

Reply:
Sure, I think they can be final